crates/services/importer/src/importer.rs

```diff
@@ -228,17 +242,18 @@ where
         let db_after_execution = db_tx.as_mut();
 
         // Importer expects that `UncommittedResult` contains the result of block
-        // execution(It includes the block itself).
+        // execution without block itself.
+        let expected_height = self.database.latest_block_height()?;
         let actual_height = db_after_execution.latest_block_height()?;
-        if expected_next_height != actual_height {
+        if expected_height != actual_height {
             return Err(Error::InvalidDatabaseStateAfterExecution(
-                expected_next_height,
+                expected_height,
                 actual_height,
             ))
         }
 
         db_after_execution
-            .seal_block(&block_id, &result.sealed_block.consensus)?
+            .block(&self.chain_id, &result.sealed_block)?
```

Comment:
`block` seems too generic - it's not clear from the name what this function actually does with the block.

```suggestion
            .store_block(&self.chain_id, &result.sealed_block)?
```